### PR TITLE
node and octocore update

### PR DESCRIPTION
**Pull request by keifgwinn**
This should update the action for set-output, and supported node version. Initial pull didn't work, apologies. I've fixed that up now, and also changed it to not require node modules to be checked in.

We're getting these alerts on the current 1.2 version.

<html>
<body>

Node.js 12 actions are deprecated. Please update the following actions to use Node.js 16: Azure/pipelines@v1.2. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12

--

The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands

</body>
</html>

### CODE REVIEW

**Comment by keifgwinn** [comment]
@microsoft-github-policy-service agree company="3D Repo"

**Comment by bartoszkosiorek** [comment]
`node12` usage is an blocker for using this action on [Apple M1 devices](https://en.wikipedia.org/wiki/Apple_M1), as the error:
```
The node12 is not supported on macOS ARM64 platform. Use node16 instead.
```
is displayed.

**Comment by keifgwinn** [comment]
> `node12` usage is an blocker for using this action on [Apple M1 devices](https://en.wikipedia.org/wiki/Apple_M1), as the error:
> 
> ```
> The node12 is not supported on macOS ARM64 platform. Use node16 instead.
> ```
> 
> is displayed.

this branch is working on my pipeline now without warnings. 

image

File: dist/package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (JamesBurnside) [inline dist/package.json:1]: Is it intentional the `dist/` folder was included in this PR?

**Comment by keifgwinn** [comment]
Yes, otherwise it requires the node_modules installed in the thing that runs the action so was simpler for me to implement that compile to a single resource that the other actions use--KeifOn 18 Apr 2023, at 17:51, James Burnside ***@***.***> wrote:﻿
@JamesBurnside commented on this pull request.



In dist/package.json:
> @@ -0,0 +1,39 @@
+{

Is it intentional the dist/ folder was included in this PR?

**Review by tjcorr — APPROVED** [review]

**Pull request merged**